reps/2022-12-08-ray-for-federated-learning-and-privacy-preserving-computing.md

```diff
@@ -0,0 +1,172 @@
+
+## Summary - Ray for federated learning and privacy-preserving computing
+### General Motivation
+Federated machine learning and privacy-preserving computing are getting wider. They're obviously distributed systems because they're always across parties(companies or ORGs.) To extend Ray ecosystem to federated learning and privacy-preserving computing domain, we should let users be easy to use Ray, for building their own federated learning and privacy-preserving computing applications without any concern on data privacy, task-attacks, illegal invasion, and etc.
+
+PS: I also found another related issue: https://github.com/ray-project/ray/issues/25846
+
+In this proposal, we'd like to build a connector layer on Ray, to provide the ability for users use Ray easily to build their such kind of system, with the secure issues addressed totally.
+
+### Key requirements:
+- Provide the ability for users to easily build their such kind of application on Ray.
+- Setup different Ray clusters for different parties to avoid uncotrollable complex Ray protocols. 
+  - We have tried to setup a single cross-party Ray cluster but failed for security reasons. It is difficult for parties to detect and prevent attacks, e.g. a malicious attacker (it could even be one of the parties) runs destructive code. The complex Ray protocols make it very difficult to enhance the security under single Ray cluster. 
+- Have a unified and global-viewed programming mode for different parties.
+- Data transmition across parties should be in push mode instead of pull mode. Pull mode makes it hard to prevent malicious attacker stealing data. So push mode is much better since it's the responsibility of the party to decide whether to send data to others.
```

Comment:
```suggestion
- Data transmission across parties should be in push mode instead of pull mode. It's hard to prevent malicious attacker stealing data in pull mode. Push mode is much better since it's the responsibility of the party to decide whether to send data to others.
```